### PR TITLE
feat: node-bindings alloy api example

### PR DESCRIPTION
**Pull request by pawurb**
## Motivation

This example shows how to mock storage values after forking with Anvil. It's useful for running local simulations. 

## PR Checklist

- [x] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

**Review by yash-atreya — APPROVED** [review]
lgtm!

File: examples/node-bindings/examples/anvil_set_storage_at.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+//! Example of mocking WETH balance of a target account
```
Comment (zerosnacks) [inline examples/node-bindings/examples/anvil_set_storage_at.rs:1]:
```suggestion
//! Example of mocking WETH balance of a target account using `anvil_set_storage_at`.
```

File: examples/node-bindings/examples/anvil_set_storage_at.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    println!("WETH balance before: {}", balance_before);
+
+    // Mock WETH balance using the Anvil API.
+    let hashed_slot = keccak256((account, U256::from(3)).abi_encode());
+    let one_ether = U256::from(1_000_000_000_000_000_000u128);
+    provider.anvil_set_storage_at(WETH_ADDR, hashed_slot.into(), one_ether.into()).await?;
+
+    let balance_after = iweth.balanceOf(account).call().await?._0;
+    println!("WETH balance after: {}", balance_after);
+
```
Comment (zerosnacks) [inline examples/node-bindings/examples/anvil_set_storage_at.rs:48]: I like adding `assert_eq!` here as well, mainly to check for regressions

File: examples/node-bindings/examples/anvil_set_storage_at.rs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    // Spin up a forked Anvil node.
+    // Ensure `anvil` is available in $PATH.
+    let anvil = Anvil::new().fork(fork_url).try_spawn()?;
+    let provider =
+        ProviderBuilder::new().with_recommended_fillers().on_http(anvil.endpoint().parse()?);
+
+    let iweth = IERC20::new(WETH_ADDR, provider.clone());
+
+    // random empty account
+    let account = address!("F605F9d1cB055E87E30bcAEe4CB9389a35aBe8Ff");
```
Comment (zerosnacks) [inline examples/node-bindings/examples/anvil_set_storage_at.rs:36]:
```suggestion
    // Random empty account.
    let account = address!("F605F9d1cB055E87E30bcAEe4CB9389a35aBe8Ff");
```

**Review by zerosnacks — APPROVED** [review]

File: examples/node-bindings/examples/anvil_set_storage_at.rs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    // Random empty account.
+    let account = address!("F605F9d1cB055E87E30bcAEe4CB9389a35aBe8Ff");
+
+    let balance_before = iweth.balanceOf(account).call().await?._0;
+    println!("WETH balance before: {}", balance_before);
+    assert_eq!(balance_before, U256::ZERO);
+
+    // Mock WETH balance using the Anvil API.
+    let hashed_slot = keccak256((account, U256::from(3)).abi_encode());
+    let one_ether = U256::from(1_000_000_000_000_000_000u128);
```
Comment (zerosnacks) [inline examples/node-bindings/examples/anvil_set_storage_at.rs:44]:
```suggestion
    let one_ether: U256 = parse_units("1.0", "ether")?.into();
```

File: examples/node-bindings/examples/anvil_set_storage_at.rs [hunk 5]
```diff
@@ -0,0 +1,52 @@
+//! Example of mocking WETH balance of a target account using `anvil_set_storage_at`.
+
+use alloy::{
+    node_bindings::Anvil,
+    primitives::{address, keccak256, Address, U256},
```
Comment (zerosnacks) [inline examples/node-bindings/examples/anvil_set_storage_at.rs:5]:
```suggestion
    primitives::{address, keccak256, utils::parse_units, Address, U256},
```

**Comment by zerosnacks** [comment]
Thanks!

**Pull request merged**